contracts/src/bridge/IInbox.sol

```diff
@@ -23,6 +23,10 @@ error InsufficientValue(uint256 expected, uint256 actual);
 /// @dev submission cost provided isn't enough to create retryable ticket
 error InsufficientSubmissionCost(uint256 expected, uint256 actual);
 
+/// @dev used to convey retryable tx data in eth calls without requiring a tx trace
+/// this follows a pattern similar to EIP-3668 where reverts surface call information
+error RetryableData(address sender, uint256 dataLength, bytes data);
```

Comment:
Do we need the dataLength here? I think data should contain its length

Reply:
We should also specify `to`, `callvalue`, etc.

Reply:
Specifically it seems like we should just surface all the arguments other than l2 gas limit and gas price

Reply:
Added in the missing values (including gas limit and price for consistency - there is no added cost in regular interaction and we may want to throw the same error signature in the future in a diff way).

Reply:
Added the `dataLength` here to follow the pattern used in the code block below.
The error data also gets abi encoded, so i believe that helps with the parsing of dynamic size data

Reply:
Ah the reason it's needed below is because we're using `abi.encodePacked` which isn't natively reversible rather than abi encoding. I think it's probably cleaner to drop the extra arg since it could be a bit confusing to consumers why it's there